web/src/pages/Settings/Streams/StreamModal/StreamModal.jsx

```diff
@@ -190,18 +192,35 @@ const StreamModal = ({ stream, onClose, apply, del }) => {
                     streamTemplate.fields.map((field) => {
                         switch (field.type) {
                         case "text":
-                            return (
-                                <TextField
-                                    key={field.name}
-                                    name={field.name}
-                                    desc={field.desc}
-                                    required={field.required}
-                                    defaultValue={streamFields[field.name]}
-                                    onChange={(v) => {
-                                        setStreamFields({ ...streamFields, [field.name]: v });
-                                    }}
-                                />
-                            );
+                            if(field.name == "password"){
+                                return (
+                                    <TextField
+                                        key={field.name}
+                                        name={field.name}
+                                        desc={field.desc}
+                                        type="password"
```

Comment:
can you just toggle this field on & off instead of duplicating this many lines of code and having a conditional return value?